ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -385,3 +390,112 @@ def set_upgrade_images(self):
             with NamedTemporaryFile() as cs_yaml:
                 dump_data_to_temp_yaml(cs_data, cs_yaml.name)
                 ocs_catalog.apply(cs_yaml.name)
+
+
+class UpgradeDisruptiveOps(EcosystemTest):
+    """
+    Contains disruptive upgrade tests functions
+
+    """
+
+    def get_ocs_operator_node_name(self):
+        """
+        Getting node's name that running ocs-operator pod
+
+        Returns:
+            str: node's name that running ocs-operator pod
+
+        """
+        ocs_operator_pod = get_ocs_operator_pod()
+        log.debug(f"ocs operator pod info: {ocs_operator_pod}")
+        ocs_operator_node = get_pod_node(ocs_operator_pod)
+
+        return get_node_name(ocs_operator_node)
+
+    def worker_node_shutdown(self, nodes, abrupt):
+        """
+        Shutdown worker node that running ocs-operator pod
+
+        Args:
+            abrupt: bool: True if abrupt shutdown, False for permanent shutdown
+
+        """
+        log.info(f"Abrupt {abrupt}")
+        # get ocs-operator node:
+        ocs_operator_node_name = self.get_ocs_operator_node_name()
+
+        # get workers node objects:
+        node_to_shutdown = list()
+        for node in get_typed_nodes():
+            node_name = get_node_name(node)
+            log.info(f"node: {node_name}, ocs operator node: {ocs_operator_node_name}")
+            if node_name == ocs_operator_node_name:
+                # if node_name != rook_ct_node_name:
+                node_to_shutdown.append(node)
+                log.info(f"node to shutdown: {get_node_name(node_to_shutdown[0])}")
+                nodes.stop_nodes(node_to_shutdown)
+                log.info("stop instance - done!")
+                break
+
+        log.info("Sleeping 5 minutes")
+        time.sleep(320)
+        assert wait_for_ct_pod_recovery(), "Ceph tools pod failed to come up on another node"
+        if abrupt:
+            log.info("Abrupt Shutdown")
+            if node_to_shutdown:
+                nodes.start_nodes(nodes=node_to_shutdown)
+
+
+@pytest.mark.parametrize()
```

Comment:
I am aware of parametrize of tests or fixtures, but here it looks like some hack where you are trying to parametrize regular function here. Can you please explain the logic behind that as this doesn't look like the proper use case of parametrize here. I see you already parametrize test_worker_node_shutdown TC, so why it's needed here like this?

Reply:
i need to pass 'nodes' fixture as parameter to run_ocs_upgrade function, i would love to find another way to implement it

Reply:
Yes, please set up mtg with me for afternoon ;) like 3pm your time